### PR TITLE
Declarative chart bug fixes

### PR DESCRIPTION
**Pull request by Anush2303**
Bugs fixed:
1. Heatmap: chart crashes if colorscale array not provided in data.
2. Sankey: if source and target does not contain nodes with same number or negative numbered nodes, chart crashes.
3. Horizontal bar chart: Bars are overlapping.
4. Line chart: Vertical dotted line going below x-axis

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts [hunk 1]
```diff
@@ -321,12 +321,13 @@ export const transformPlotlyJsonToHorizontalBarWithAxisProps = (
     })
```
Comment (fabricteam) [inline packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:321]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33426&commitId=f7458fed868eaa5c120e48d1f1ef01a844dd8ad8&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Gap space 25.chromium.png|2309| Changed|
|Callout.Top right edge.chromium.png|1594| Changed|
|Callout.Left center.chromium.png|2616| Changed|

</details>


  <details><summary><b>react-charting-HeatMapChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-HeatMapChart.Basic RTL.chromium.png|372| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by srmukher** [comment]
> ## Previous Behavior
> ## New Behavior
> ## Related Issue(s)
> * Fixes #

Can you give a description of what bugs were fixed?

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts [hunk 2]
```diff
@@ -321,12 +321,13 @@ export const transformPlotlyJsonToHorizontalBarWithAxisProps = (
...
     .flat();
 
-  const chartHeight = layout.height || 350;
+  const chartHeight = layout.height || 450;
```
Comment (srmukher) [inline packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:324]: how are we getting this number?

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts [hunk 3]
```diff
@@ -371,8 +372,10 @@ export const transformPlotlyJsonToHeatmapProps = (jsonObj: any): IHeatMapChartPr
   };
 
   // Convert normalized values to actual values
-  const domainValuesForColorScale: number[] = firstData.colorscale?.map((arr: any) => arr[0] * (zMax - zMin) + zMin);
-  const rangeValuesForColorScale: string[] = firstData.colorscale?.map((arr: any) => arr[1]);
+  const domainValuesForColorScale: number[] = firstData.colorscale
+    ? firstData.colorscale.map((arr: any) => arr[0] * (zMax - zMin) + zMin)
+    : [];
```
Comment (srmukher) [inline packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:377]: do we need to keep any default colorscale value for heatmap as we have default colors for other charts?

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts [hunk 4]
```diff
@@ -396,7 +399,8 @@ export const transformPlotlyJsonToSankeyProps = (
     }))
     // eslint-disable-next-line @typescript-eslint/no-shadow
     //@ts-expect-error Dynamic link object. Ignore for now.
-    .filter(x => x.source !== x.target); // Filter out self-references (circular links)
+    // Filter out non-negative nodes, unequal nodes and self-references (circular links)
+    .filter(x => x.source > 0 && x.target > 0 && x.source !== x.target);
```
Comment (srmukher) [inline packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:403]: circular links can also happen if source1 -> target1 and target1 -> source1. Are we also handling this scenario?

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts (hunk 3 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:377]: In V8, for other charts like Sankey, we do have, so just check if we need to add for Heatmap too.

Comment (Anush2303) [reply packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:377]: I meant, for heatmap we don't have

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts [hunk 5]
```diff
@@ -396,7 +399,8 @@ export const transformPlotlyJsonToSankeyProps = (
     }))
     // eslint-disable-next-line @typescript-eslint/no-shadow
     //@ts-expect-error Dynamic link object. Ignore for now.
-    .filter(x => x.source !== x.target); // Filter out self-references (circular links)
+    // Filter out non-negative nodes, unequal nodes and self-references (circular links)
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:402]: >Filter out non-negative nodes [](http://example.com/codeflow?start=7&length=29)

This is - Filter out negative nodes.

**Review by AtishayMsft — APPROVED** [review]
:shipit:

**Comment by Anush2303** [comment]
> Bugs fixed:
> 
> 1. Heatmap: chart crashes if colorscale array not provided in data.
> 2. Sankey: if source and target does not contain nodes with same number or negative numbered nodes, chart crashes.
> 3. Horizontal bar chart: Bars are overlapping.
> 4. Line chart: Vertical dotted line going below x-axis.
> 
> ## Related Issue(s)
> * Fixes #

**Pull request merged**